### PR TITLE
Feat/filter amdministrators in syncofficers

### PR DESCRIPTION
**Pull request by kajm**
## Explain the changes you’ve made
Filter out only allowed officers types that should be returned by the function

## Explain why these changes are made
There is a need of only showing `officers` working on a case and not administrators.

## Explain your solution
Created a helper function which filters out only allowed officers by their type by a allowed list.

## How to test

1. Checkout this branch
2. run `sls deploy` within service folder for `service viva-ms`
3. .. or run tests

## Tested environments

- [] Your personal AWS environment.
- [x] Your local Serverless environment.

### CODE REVIEW

File: services/viva-ms/src/lambdas/syncOfficers.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import vivaAdapter from '../helpers/vivaAdapterRequestClient';
+import officers from '../helpers/officers';
+
+import { CaseAdministrator } from '../types/caseItem';
+import { VivaOfficer, VivaOfficerType } from '../types/vivaMyPages';
+import { CaseItem } from '../types/caseItem';
+
+type CaseKeys = Pick<CaseItem, 'SK' | 'PK'>;
+
+const allowedOfficerTypes: VivaOfficerType[] = ['officer'];
```
Comment (dannil76) [inline services/viva-ms/src/lambdas/syncOfficers.ts]: Kan detta vara en enums?

File: services/viva-ms/src/lambdas/syncOfficers.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  const { PK, SK, details } = unMarshalledCaseData as CaseItem;
+  const administrators = details?.administrators ?? [];
+
+  const personalNumber = PK.substring(5);
+  const getOfficersResult = await getVivaOfficers(personalNumber);
+
+  const { officer } = getOfficersResult;
+  const parsedVivaOfficers = officers.parseVivaOfficers(officer);
+  const filteredVivaOfficers = parsedVivaOfficers.filter(officer =>
+    officers.filterVivaOfficerByType(officer, allowedOfficerTypes)
```
Comment (dannil76) [inline services/viva-ms/src/lambdas/syncOfficers.ts]: Lite otydligt om denna utbrytningen är nödvänding?
Kanske kan lösas med lista av enums?

File: services/viva-ms/src/types/vivaMyPages.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+export type VivaOfficerType = 'officer' | 'assistant';
```
Comment (dannil76) [inline services/viva-ms/src/types/vivaMyPages.ts]: Enums?

**Review by dannil76 — CHANGES_REQUESTED** [review]

**Comment by helsingborgsstad** [comment]
Task linked: [CU-2t367gf Filtrera ut Socialsekreterare i synkofficers](https://app.clickup.com/t/2t367gf)

File: services/viva-ms/src/lambdas/syncOfficers.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  const { getVivaOfficers, updateCaseAdministrators } = dependencies;
+
+  const unMarshalledCaseData = dynamoDbConverter.unmarshall(input.detail.dynamodb.NewImage);
+
+  const { PK, SK, details } = unMarshalledCaseData as CaseItem;
+  const administrators = details?.administrators ?? [];
+
+  const personalNumber = PK.substring(5);
```
Comment (MasterKenth) [inline services/viva-ms/src/lambdas/syncOfficers.ts:65]: `.substring(5)` seems very magical 🧙‍♂️. (DNF)

**Review by MasterKenth — CHANGES_REQUESTED** [review]

File: services/viva-ms/src/lambdas/syncOfficers.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    Key: keys,
+    UpdateExpression: 'SET details.administrators = :newAdministrators',
+    ExpressionAttributeValues: { ':newAdministrators': newAdministrators },
+    ReturnValues: 'NONE',
+  };
+
+  return dynamoDb.call('update', dynamoDbParams);
+}
+
+const dynamoDbConverter = AWS.DynamoDB.Converter;
```
Comment (dannil76) [inline services/viva-ms/src/lambdas/syncOfficers.ts:33]: Don't we have an helper for this? :)

File: services/viva-ms/src/lambdas/syncOfficers.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+export async function syncOfficers(input: LambdaRequest, dependencies: Dependencies) {
+  if (input.detail.dynamodb.NewImage === undefined) {
+    return null;
+  }
+
+  const { getVivaOfficers, updateCaseAdministrators } = dependencies;
+
+  const unMarshalledCaseData = dynamoDbConverter.unmarshall(input.detail.dynamodb.NewImage);
```
Comment (dannil76) [inline services/viva-ms/src/lambdas/syncOfficers.ts:60]: Think we have an helper in our toolbox

File: services/viva-ms/src/lambdas/syncOfficers.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  const administrators = details?.administrators ?? [];
+
+  const personalNumber = PK.substring(5);
+  const getOfficersResult = await getVivaOfficers(personalNumber);
+
+  const { officer } = getOfficersResult;
+  const parsedVivaOfficers = officers.parseVivaOfficers(officer);
+  const filteredVivaOfficers = parsedVivaOfficers.filter(({ type }) =>
+    allowedOfficerTypes.includes(type)
+  );
```
Comment (dannil76) [inline services/viva-ms/src/lambdas/syncOfficers.ts:72]: This is somewhat hard to understand. Can it be simplified?

**Review by dannil76 — CHANGES_REQUESTED** [review]

File: services/viva-ms/test/helpers/officers.test.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+describe('parseVivaOfficers', () => {
+  it('successfully parses a viva officers array to `CaseAdministrator` format', () => {
+    const expectedResult = [getAdministratorObject({})];
+
+    const result = officers.parseVivaOfficers([
+      {
+        mail: 'email',
+        name: 'name',
```
[the remark this answers is not in the document]
Comment (kajm) [reply services/viva-ms/test/helpers/officers.test.ts:28]: Will not change in this PR. Another task will be created instead

File: services/viva-ms/src/lambdas/syncOfficers.ts (hunk 7 above)
Comment (kajm) [reply services/viva-ms/src/lambdas/syncOfficers.ts:72]: I think its easy enough to understand. Will not change

File: services/viva-ms/src/lambdas/syncOfficers.ts (hunk 4 above)
Comment (kajm) [reply services/viva-ms/src/lambdas/syncOfficers.ts:63]: Will not change in this PR. Another task will be created instead

**Review by dannil76 — APPROVED** [review]

**Pull request merged**